File: oeuvre.php
<?php
    require_once 'header.php';
    require_once 'database.php';

    // Si l'URL ne contient pas d'id, on redirige sur la page d'accueil
    if(empty($_GET['id'])) {
        header('Location: index.php');
    }

    $db = DBConnection();
    $stmt = $db->prepare('SELECT * FROM oeuvres WHERE id = :id');
    $isSuccess = $stmt->execute([ ':id' => intval($_GET['id']) ]);

    if(!$isSuccess) {
        header('Location: index.php');
        exit();
    }

    $oeuvre = $stmt->fetch(\PDO::FETCH_ASSOC);

    // Si aucune oeuvre trouvé, on redirige vers la page d'accueil
    if(!$oeuvre) {
        header('Location: index.php');
        exit();
    }
?>

<article id="detail-oeuvre">
    <div id="img-oeuvre">
        <img src="<?= $oeuvre['image'] ?>" alt="<?= $oeuvre['titre'] ?>">
    </div>
    <div id="contenu-oeuvre">
        <h1><?= $oeuvre['titre'] ?></h1>
        <p class="description"><?= $oeuvre['artiste'] ?></p>
        <p class="description-complete">
             <?= $oeuvre['description'] ?>
        </p>
    </div>
</article>

<?php require_once 'footer.php'; ?>

File: go/oeuvre.go
package main

type oeuvre struct {
	Id          int
	Titre       string
	Description string
	Artiste     string
	Image       string
}

func (a *artBox) GetOeuvres() ([]oeuvre, error) {
	oeuvres := []oeuvre{}

	rows, err := a.db.Query("SELECT * FROM oeuvres")
	if err != nil {
		return oeuvres, err
	}
	defer rows.Close()

	for rows.Next() {
		o := oeuvre{}
		rows.Scan(
			&o.Id,
			&o.Titre,
			&o.Description,
			&o.Artiste,
			&o.Image,
		)

		oeuvres = append(oeuvres, o)
	}

	return oeuvres, nil
}
